lib/ruby/stdlib/date.rb

```diff
@@ -482,6 +482,9 @@ def self.strptime(str='-4712-01-01', fmt='%F', sg=ITALY)
   #
   # +sg+ specifies the Day of Calendar Reform.
   def self.parse(str='-4712-01-01', comp=true, sg=ITALY)
+    unless str.kind_of?(::String) || str.respond_to?(:to_str)
```

Comment:
could we just move this to `_parse` ? 
and than it wouldn't need to exist on 3 places + MRI seems to validate there as well:
```
2.5.1 :003 > Date._parse 111
Traceback (most recent call last):
        3: from /opt/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        2: from (irb):3
        1: from (irb):3:in `_parse'
TypeError (no implicit conversion of Integer into String)
```
... and since this would eventually be moved to native like they did it will be simpler for anyone ending up doing it.